src/ply.rs

```diff
@@ -246,60 +292,75 @@ fn open(ply_file: &Path) -> Result<(BufReader<File>, i64, PointFormat)> {
     }
 
     let vertex = &header["vertex"];
-    if !vertex.has_property("x") || !vertex.has_property("y") || !vertex.has_property("z") {
-        panic!("PLY must contain properties 'x', 'y', 'z' for 'vertex'.");
-    }
+    let mut seen_x = false;
+    let mut seen_y = false;
+    let mut seen_z = false;
 
+    let mut readers: Vec<ReadingFunc> = Vec::new();
     let mut num_bytes_per_point = 0;
-    let position_data_type = header["vertex"]["x"].data_type;
-    num_bytes_per_point += position_data_type.size_in_bytes() * 3;
-
-    // TODO(hrapp): We silently assume that data types of (y, z) is that of x and similar for
-    // colors. Track this properly.
-    let color_data_type = {
-        if vertex.has_property("red") && vertex.has_property("green") &&
-           vertex.has_property("blue") {
-            Some(vertex["red"].data_type)
-        } else if vertex.has_property("r") && vertex.has_property("g") && vertex.has_property("b") {
-            Some(vertex["r"].data_type)
-        } else {
-            None
+
+    for prop in &vertex.properties {
+        match &prop.name as &str {
+            "x" => {
+                readers.push(read_casted_property!(prop.data_type,
+                                                   point.position.x,
+                                                   num_bytes_per_point));
+                seen_x = true;
+            }
+            "y" => {
+                readers.push(read_casted_property!(prop.data_type,
+                                                   point.position.y,
+                                                   num_bytes_per_point));
+                seen_y = true;
+            }
+            "z" => {
+                readers.push(read_casted_property!(prop.data_type,
+                                                   point.position.z,
+                                                   num_bytes_per_point));
+                seen_z = true;
+            }
+            "r" | "red" => {
+                readers.push(read_casted_property!(prop.data_type, point.r, num_bytes_per_point));
+            }
+            "g" | "green" => {
+                readers.push(read_casted_property!(prop.data_type, point.g, num_bytes_per_point));
+            }
+            "b" | "blue" => {
+                readers.push(read_casted_property!(prop.data_type, point.b, num_bytes_per_point));
+            }
+            other => println!("Ignoring property '{}' on 'vertex'.", other),
```

Comment:
This doesn't seem to actually ignore data, but rather read malformed data. Add a reader that ignores or panic instead?